### ISSUE TITLE
Okhttp version is old in quarkus-universe-bom

### ISSUE THREAD

**Issue by mmaryo** [quarkusio/quarkus#12680]
I'm facing this error https://github.com/square/okhttp/issues/5818
And realize I have 2 okhttp versions 

One from quarkus-universe-bom (version 3.x.x)
One from org.web3j (version 4.x.x=)

There are breaking change between that versions

Have you plan to upgrade okhttp ?

**Comment by mmaryo** [comment]
also the same for web3j
from quarkus-universe-bom (version 3.6.0)
lastest org.web3j (version 4.7.0)

**Comment by mmaryo** [comment]
Can you update web3j version ?

That make I cannot use Quarkus because bom contains very old web3 version 

https://github.com/web3j/web3j/releases

https://oss.sonatype.org/service/local/repositories/releases/content/io/quarkus/quarkus-universe-bom/1.9.2.Final/quarkus-universe-bom-1.9.2.Final.pom

**Comment by gsmet** [comment]
@vsevel IIRC okhttp is used by Vault? Any chance we could upgrade there?

@mmaryo you can simply override the dependency version in your local project.

**Comment by vsevel** [comment]
> IIRC okhttp is used by Vault? Any chance we could upgrade there?

yes. sure.
it is being used by the kubernetes client extension as well. it has a dependency on an even older version [3.12.12](https://github.com/fabric8io/kubernetes-client/blob/master/pom.xml#L83)
so I can make a quick try, but I have no idea if the k8s client will support it beside vault.
may be @gastaldi would know.

**Comment by vsevel** [comment]
I used okhttp client 4.9.0
vault is ok, but I have some errors with the kubernetes client:
```
Caused by: java.lang.NoSuchMethodError: okhttp3.internal.Internal.initializeInstanceForTests()V
        at okhttp3.mockwebserver.MockWebServer.<clinit>(MockWebServer.java:103)
        at io.fabric8.kubernetes.client.server.mock.KubernetesMockServer.<init>(KubernetesMockServer.java:49)
        at io.quarkus.test.kubernetes.client.KubernetesMockServerTestResource.createMockServer(KubernetesMockServerTestResource.java:37)
        at io.quarkus.test.kubernetes.client.KubernetesMockServerTestResource.start(KubernetesMockServerTestResource.java:25)
        at io.quarkus.test.common.TestResourceManager.start(TestResourceManager.java:52)
```
we would need an update in the fabric8 project first before we change the version in the bom.

**Comment by gastaldi** [comment]
I can have a look, but I am not so confident that would happen soon (/cc @iocanel @geoand).

Any chance to have https://github.com/quarkusio/quarkus/issues/4162 fixed soon?

**Comment by gsmet** [comment]
Well, moving Vault away from Okhttp won't change a thing if the Kubernetes client uses it: it will still be in the bom.

So we will definitely need an upgrade to the Kubernetes Client.

BTW, I was pleasantly surprised by okhttp, it works perfectly in native without any changes.

**Comment by geoand** [comment]
Yeah, OkHTTP is great

**Comment by gastaldi** [comment]
I love OkHTTP, the only sad thing is that their latest version is written in Kotlin and that requires the Kotlin runtime to run, making the dist bigger

**Comment by vsevel** [comment]
when I initially worked on vault I started with the apache httpclient, but I endep up choosing okhttp client because that's what the kubernetes client was doing already, so at least I was bringing in another lib.
and the way I had to approach the config source did not allow me to use the rest client as explained here https://github.com/quarkusio/quarkus/issues/4162.
but with progress on https://github.com/quarkusio/quarkus/pull/13498, may be switching to the quarkus rest client, or another quarkus option (i.e. Vert.x/Mutiny, such as this attempt https://github.com/quarkusio/quarkus/pull/8885).
the java 11 HTTP Client API might be an option as well if support for java 8 could be dropped.

**Comment by geoand** [comment]
Closing as won't fix

**Comment by jtama** [comment]
@geoand  why exactly won't the okhttp version be updated ? https://github.com/quarkiverse/quarkus-minio/actions/runs/1047560127 is caused by this deprecated version. Is there any thing I can help to update the version ?

**Comment by gastaldi** [comment]
Because AFAIK the Kubernetes Client extension needs the older version and won't work with the latest and greatest (see  https://github.com/quarkusio/quarkus/issues/12680#issuecomment-734410520).

**Comment by jtama** [comment]
The kubernetes java client has been updated. Could we hope a progress on this now ? 
https://github.com/kubernetes-client/java/issues/2095

**Comment by geoand** [comment]
Not sure what you are asking for, can you elaborate please?

**Comment by jtama** [comment]
The okhttp version used in quarkus is really old. Until now we couldn't update it because it was used by the kubernetes java client. Now that the lib has updated it's okhttp dependency, can we hope to do so in the universe bom ?

The minio extension depends on the minio sdk, but. we are stuck in an obsolete version because of this okhttp version mismatch.

**Comment by geoand** [comment]
> The okhttp version used in quarkus is really old. Until now we couldn't update it because it was used by the kubernetes java client. Now that the lib has updated it's okhttp dependency, can we hope to do so in the universe bom ?

We use the [Fabric8 Kubernetes Client](https://github.com/fabric8io/kubernetes-client) which is not going to be upgraded to okhttp `4.x`

**Comment by gsmet** [comment]
@geoand any news on the move to the Vert.x client? I can feel the pain of the users as I also started having compatibility issues between what libraries use and what we enforce (typically the version we enforce can't be used anymore with the GitHub API).

**Comment by geoand** [comment]
AFAIK, the first step of introducing an abstraction layer is done and the next step is to move to JDK 11's HTTP Client. Using the Vert.x client is likely something going to be done in the midterm...

I absolutely understand the issue with the old OkHTTP version and hopefully sometime soon we'll be able to  completely remove it from the BOM so users can use whatever version they need.

**Comment by jtama** [comment]
And is there anything we can do to help on the subject ?

**Comment by geoand** [comment]
Not that I know of, but @manusa would know more

**Comment by manusa** [comment]
The underlying HTTP client implementation (currently OkHTTP) will be completely abstracted away in v6.0.0 of the Fabric8 Kubernetes Client, as Georgios mentioned.

@shawkins has done an excellent job on this initiative. The initial step to provide a httpclient-less API for the client is in https://github.com/fabric8io/kubernetes-client/pull/3654, which should be merged by the end of this week.

Next steps are introducing a Java native Http client implementation (https://github.com/shawkins/httpclient).
Providing a Vet.x implementation should be doable to once this initial PR is merged. @metacosm mentioned that maybe someone from the Vert.x team could help out for this task.

In any case, currently, users should be technically able to replace the provided OkHttp version with a [more recent 4.x version](https://github.com/fabric8io/kubernetes-client/issues/2632#issuecomment-748434878).

**Comment by lazystone** [comment]
So, can this issue be re-opened then and OkHttp finally upgraded?

**Comment by geoand** [comment]
The plan is to remove OkHTTP from the BOM completely when we move to a version of Fabric8 Kubernetes Client that does not include OkHTTP

**Comment by SpeedsterF2** [comment]
As of commit f1a50bfe683ea075f67e29321d84508433ed7c63 the Fabric8 Kubernetes is 6.x, but the bom still has

https://github.com/quarkusio/quarkus/blob/2.14.3.Final/bom/application/pom.xml
...
        <kubernetes-client.version>6.2.0</kubernetes-client.version> 
...
        <okhttp.version>3.14.9</okhttp.version>
        <okio.version>1.17.2</okio.version>
        
I am stuck with old minio version now, too.

**Comment by gsmet** [comment]
The switch to the Vert.x HTTP client for the Kubernetes Client in Quarkus is still a work in progress.

**Comment by SpeedsterF2** [comment]
> The switch to the Vert.x HTTP client for the Kubernetes Client in Quarkus is still a work in progress.

But as far as I understand blocking the update of okhttp was the fabricio dependency, which made the jump to 6x some time ago? Or did I miss something?

FYI, here my PR on the minio extension in quarkiverse:
https://github.com/quarkiverse/quarkus-minio/pull/183

**Comment by geoand** [comment]
> But as far as I understand blocking the update of okhttp was the fabricio dependency, which made the jump to 6x some time ago? Or did I miss something?

https://github.com/fabric8io/kubernetes-client/blob/v6.2.0/pom.xml#L84

**Comment by SpeedsterF2** [comment]
Thanks gsmet and geoand for awnsering to my comments.

**Comment by geoand** [comment]
🙏🏼

**Comment by gsmet** [comment]
@SpeedsterF2 the idea is to get rid of Okhttp in Quarkus and use the Vert.x HTTP client instead (Kubernetes Client 6.x added an abstraction layer for that). IIRC, there were still a few issues left before we could switch to this new Vert.x based client.

@manusa will know more.

**Comment by SpeedsterF2** [comment]
@gsmet
So unless someone alters the minio client lib to be able to handle different HTTP back-ends, minio and quarkus native image will have issues. 😭 
I even failed to just define a @RegisterForReflection to mimic what is inside of the MinioNativeProcessor
https://github.com/quarkiverse/quarkus-minio/blob/main/minio-native/deployment/src/main/java/io/quarkiverse/minio/client/deployment/MinioNativeProcessor.java
All the *Label classes are not even public!

**Comment by SpeedsterF2** [comment]
https://mvnrepository.com/artifact/com.carrotsearch.thirdparty/simple-xml-safe
vs.
https://mvnrepository.com/artifact/org.simpleframework/simple-xml/2.7.1
